modules/token-holders/pages/self-custody.adoc

```diff
@@ -0,0 +1,111 @@
+
+If you need help meeting any of these prerequisites, see link:newcomers{outfilesuffix}[Preliminary steps for newcomers].
+If you are comfortable meeting the prerequisites without additional instructions, continue to <<Download and install>>.
+
+[[download-and-install]]
+== Download and install
+
+You can download the latest version of the {sdk-long-name} directly from within a terminal shell on your local computer.
+
+To download and install:
+
+[arabic]
+. Open a terminal shell on your local computer.
++
+For example, on macOS, open Applications, Utilities, then double-click *Terminal* or press kbd:[{commandkey} + spacebar] to open Search, then type `terminal`.
+. Download and install the {sdk-short-name} package by running the following command:
++
+[source,bash]
+----
+sh -ci "$(curl -fsSL https://sdk.dfinity.org/install.sh)"
+----
++
+This command prompts you to read and accept the license agreement before installing the {company-id} execution command-line interface (CLI) and its dependencies on your local computer.
+. Type `+y+` and press kbd:[Return] to continue with the installation.
++
+After you accept the license agreement, the installation script installs components on the local computer then returns to the command-line prompt.
+. (Optional) Verify the {sdk-short-name} command-line interface (CLI) installed and the `+dfx+` executable is available in your PATH by running the following command:
++
+[source,bash]
+----
+dfx --version
+----
+
+== Generate a public/private key pair
+
+If you have a hardware security module (HSM) appliance or a hardware wallet, you can use those devices to generate and store a public/private key pair for your identity in a secure way.
+After you generate the public/private key pair you can import your certificate to create a secure identity on a connected device.
+
+If you don't have a hardware security module or a hardware wallet, you can use `+dfx+` to automatically generate and store a public/private key pair for your default identity on the local computer.
+
+== Get the principal for your identity
+
+After you have generated the public/private key pair, you can use dfx to retrieve a principal identifier. A principal identifier is a textual representation of your identity
+that is similar to an address in Bitcoin or Ethereum. The principal identifier enables the secret key associated with a user’s public key to be used to sign messages and therefore authenticate actions on your behalf. 
+
+To get the principal:
+
+. Open a terminal shell on your local computer, if you don’t already have one open.
+. Check the active identity context by running the following command:
++
+[source,bash]
+----
+dfx identity whoami
+----
+. Get the principal for you currently-active identity by running the following command:
++
+[source,bash]
+----
+dfx identity get-principal
+----
++
+This command displays the textual representation of your principal as plain text that you can copy and paste.
```

Comment:
```suggestion
This command displays the textual representation of your principal as plain text that you can copy-and-paste.
```
But if you follow https://ell.stackexchange.com/questions/3748/can-i-say-copied-and-pasted, it is just `copy-paste`. Sounds strange to me.